example-crates/wallet_esplora_blocking/src/main.rs

```diff
@@ -30,22 +33,32 @@ fn main() -> Result<(), anyhow::Error> {
     println!("Generated Address: {}", address);
 
     let balance = wallet.get_balance();
-    println!("Wallet balance before syncing: {} sats", balance.total());
+    println!("Wallet balance before syncing: {}", balance.total());
 
     print!("Syncing...");
-    let client =
-        esplora_client::Builder::new("https://blockstream.info/testnet/api").build_blocking();
-
-    let request = wallet.start_full_scan().inspect_spks_for_all_keychains({
-        let mut once = BTreeSet::<KeychainKind>::new();
-        move |keychain, spk_i, _| {
-            match once.insert(keychain) {
-                true => print!("\nScanning keychain [{:?}]", keychain),
-                false => print!(" {:<3}", spk_i),
+    let client = esplora_client::Builder::new("https://mempool.space/testnet/api").build_blocking();
+
+    fn generate_inspect(kind: KeychainKind) -> impl FnMut(u32, &Script) + Send + Sync + 'static {
+        let mut once = Some(());
+        let mut stdout = std::io::stdout();
+        move |spk_i, _| {
+            match once.take() {
+                Some(_) => print!("\nScanning keychain [{:?}]", kind),
```

Comment:
This should fix the print for `spk0`

```suggestion
                Some(_) => print!("\nScanning keychain [{:?}] {:<3}", kind, spk_i),
```

Reply:
Done in eeb3b7cc797db51b40911a5cecc9dd66edc51fb9